Makefile

```diff
@@ -0,0 +1,40 @@
+include .env
+
+CONTAINER_ENGINE ?= $(shell command -v podman 2> /dev/null || echo docker)
+COMPOSE_ENGINE ?= $(shell command -v podman-compose 2> /dev/null || echo docker-compose)
+MY_ID ?= $(shell id -u)
+CHANGE ?= New migration
+
+.PHONY: server
+
+# rebuild server image (for compose) with new and updated dependencies
+rebuild-server:
+	$(COMPOSE_ENGINE) build --no-cache server
+
+server-up:
+	$(COMPOSE_ENGINE) up --build --force-recreate -d server
+
+# run alembic revision in another pod
+alembic-generate-revision: server-up
+	@echo "Waiting for server to start and invoking alembic upgrade head..."
+	sleep 3
+
+	@echo "Checking if database is ready..."
+	$(CONTAINER_ENGINE) run --rm --network logdetective_default \
+		quay.io/sclorg/postgresql-15-c9s \
+		pg_isready -h postgres -U $(POSTGRESQL_USER) -d $(POSTGRESQL_DATABASE) \
+		|| (echo "Database not ready -h postgres -U $(POSTGRESQL_USER) -d $(POSTGRESQL_DATABASE)"; exit 1)
+	
+	$(CONTAINER_ENGINE) run --rm -ti --user $(MY_ID) --uidmap=$(MY_ID):0:1 --uidmap=0:1:999 \
+		-e POSTGRESQL_USER=$(POSTGRESQL_USER) \
+		-e POSTGRESQL_PASSWORD=$(POSTGRESQL_PASSWORD) \
+		-e POSTGRESQL_HOST=postgres \
+		-e POSTGRESQL_DATABASE=$(POSTGRESQL_DATABASE) \
+		-v $(PWD)/alembic:/src/alembic:rw,z \
+		-v $(PWD)/alembic.ini:/src/alembic.ini:ro,z \
+		--network logdetective_default \
+		localhost/logdetective/runtime:latest \
+		bash -c "cd /src && alembic revision -m \"$(CHANGE)\" --autogenerate"
+
+	@echo "Shutting down server..."
+	$(COMPOSE_ENGINE) down
```

Comment:
this is kind of unexpected to me

Reply:
The prerequisite of this makefile's target is `podman-compose us server` , because it is the server starting file that calls `alembic upgrade head`, and here I am cleaning things. Probably I should improve it with `podman-compose down server`. Is this what you meant?

Reply:
kind of yes

my point was: if nothing is running and I want to just create a migration, then this is 100% expected - stand up the server, create migration, bring it down

but if it's already running, then it will be brought down which would be unexpected to me, hence I would comment this section out if I wanna keep the server up

this is definitely a nit pick, I'm not sure which of those 2 cases will be more common

if you propose this and feel like this is the right thing to do, let's keep it; I just wanted to be sure we understand each other's thinking process